### PR TITLE
Fix more lint errors from goreportcard

### PR DESCRIPTION
**Pull request by wongni**
*Issue #, if available:*

*Description of changes:*
Fix misspellings
Fix a few comments to satisfy golint checker
Pull getAffinityGroupIDs function to reduce cyclomatic complexity of GetOrCreateVMInstance

Somehow I was not able to buile goreportcard-cli so I used the following method to run the checks locally. (https://github.com/gojp/goreportcard/issues/379)
```
❯ docker run -it --rm -v ~/dev/pike/cluster-api-provider-cloudstack:/cluster-api-provider-cloudstack  docker.io/jenner/goreportcard-cli:latest /bin/sh -c "cd /cluster-api-provider-cloudstack && goreportcard-cli -v"
```

Current report from this PR
```
Grade: A+ (99.2%)
Files: 35
Issues: 1
gofmt: 100%
go_vet: 97%
        ../go/pkg/mod/k8s.io/client-go@v0.22.2/plugin/pkg/client/auth/exec/metrics.go
                Line 21: error: package io/fs is not in GOROOT (/usr/local/go/src/io/fs) (vet)
gocyclo: 100%
golint: 100%
ineffassign: 100%
license: 100%
misspell: 100%
```

*Testing performed:*
Full E2E test results
Ran 13 of 14 Specs in 5094.519 seconds
SUCCESS! -- 13 Passed | 0 Failed | 0 Pending | 1 Skipped

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by rejoshed — APPROVED** [review]

File: api/v1beta1/cloudstackmachine_webhook.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ import (
 // log is for logging in this package.
 var cloudstackmachinelog = logf.Log.WithName("cloudstackmachine-resource")
 
+// SetupWebhookWithManager creates a new webhook managed by passed manager
```
Comment (rejoshed) [inline api/v1beta1/cloudstackmachine_webhook.go:35]: Can you please add periods to the end of these comments? I'd like us to stick to the prevailing conventions.

File: pkg/cloud/affinity_groups.go [hunk 2]
```diff
@@ -21,17 +21,14 @@ import (
...
-)
-
+// AffinityGroup type
 type AffinityGroup struct {
 	Type string
 	Name string
 	ID   string
 }
 
+// AffinityGroupIface contains the collection of functions for AffinityGroup
```
Comment (rejoshed) [inline pkg/cloud/affinity_groups.go:31]: Here too, and probably elsewhere.

File: pkg/cloud/affinity_groups_test.go [hunk 3]
```diff
@@ -14,13 +14,13 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package cloud_test
+package cloud
```
Comment (rejoshed) [inline pkg/cloud/affinity_groups_test.go:17]: No! We separate this on purpose.

cloud_test being a separate module helps us to detect code smell in un-testable regions.

It also helps us to pair down what libs actually make it into our final module.

Why was this change made?

File: pkg/cloud/affinity_groups_test.go [hunk 4]
```diff
@@ -32,22 +32,22 @@ var _ = Describe("AffinityGroup Unit Tests", func() {
 		mockCtrl    *gomock.Controller
 		mockClient  *cloudstack.CloudStackClient
 		ags         *cloudstack.MockAffinityGroupServiceIface
-		fakeAG      *cloud.AffinityGroup
+		fakeAG      *AffinityGroup
```
Comment (rejoshed) [inline pkg/cloud/affinity_groups_test.go:35]: Yeah, not great.

File: pkg/cloud/affinity_groups_test.go [hunk 5]
```diff
@@ -47,7 +48,7 @@ var _ = Describe("AffinityGroup Unit Tests", func() {
 		client = cloud.NewClientFromCSAPIClient(mockClient)
 		fakeAG = &cloud.AffinityGroup{
 			Name: "FakeAffinityGroup",
-			Type: cloud.AffinityGroupType}
+			Type: "host affinity"}
```
Comment (rejoshed) [inline pkg/cloud/affinity_groups_test.go:51]: Please revert.

File: pkg/cloud/affinity_groups.go [hunk 6]
```diff
@@ -21,17 +21,14 @@ import (
 	"github.com/pkg/errors"
 )
 
-const (
```
Comment (rejoshed) [inline pkg/cloud/affinity_groups.go:24]: Please revert this too. Mark intentionally made these consts. 

I don't particularly care whether there are string literals in the tests or not, but these changes will only aggravate the rebase I will have to do to brink in my dummies module.

File: pkg/cloud/affinity_groups_test.go [hunk 7]
```diff
@@ -89,7 +90,7 @@ var _ = Describe("AffinityGroup Unit Tests", func() {
 			if connectionErr != nil { // Only do these tests if an actual ACS instance is available via cloud-config.
 				Skip("Could not connect to ACS instance.")
 			}
-			arbitraryAG = &cloud.AffinityGroup{Name: "ArbitraryAffinityGroup", Type: cloud.AffinityGroupType}
+			arbitraryAG = &cloud.AffinityGroup{Name: "ArbitraryAffinityGroup", Type: "host affinity"}
```
Comment (rejoshed) [inline pkg/cloud/affinity_groups_test.go:93]: Please revert.

**Comment by jweite-amazon** [comment]
I'm good with this all at my level, pending resolution of Josh's more detailed points.

**Comment by wongni** [comment]
Created https://github.com/aws/cluster-api-provider-cloudstack/pull/52 from the latest main branch.

**Pull request closed without merging**